pkg/labels/applicator.go

```diff
@@ -54,6 +54,11 @@ func (l Labeled) SameAs(o Labeled) bool {
 	return l.ID == o.ID && l.LabelType == o.LabelType
 }
 
+type WatchResult struct {
+	Labeled []Labeled
+	Valid   bool // Will be false if the underlying watcher has terminated
```

Comment:
`range` and `ok` would both be ways to check that the value is valid, but I assume most clients of this code will be using it in a `select` statement, given that it's a watch that can be interrupted. 

I know it's maybe a bit paranoid, but I really don't want clients accidentally treating an empty result the same as a nil result set.